client/dashboard/sites/index.tsx

```diff
@@ -161,25 +162,37 @@ const DEFAULT_VIEW: View = {
 	page: 1,
 	perPage: 10,
 	sort: { field: 'name', direction: 'asc' },
+	search: '',
 };
 
 export default function Sites() {
-	const navigate = useNavigate();
+	const navigate = useNavigate( { from: sitesRoute.fullPath } );
 	const sites = useQuery( sitesQuery() ).data;
 	const hasA8CSites = sites?.some( ( site ) => site.is_a8c );
-	const [ view, setView ] = useState< View >(
-		hasA8CSites
-			? {
-					...DEFAULT_VIEW,
-					filters: [
-						{
-							field: 'is_a8c',
-							operator: 'is',
-							value: false,
-						},
-					],
-			  }
-			: DEFAULT_VIEW
+	const defaultView = useMemo(
+		() =>
+			hasA8CSites
+				? {
+						...DEFAULT_VIEW,
+						filters: [
+							{
+								field: 'is_a8c',
+								operator: 'is' as Operator,
+								value: false,
+							},
+						],
+				  }
+				: DEFAULT_VIEW,
+		[ hasA8CSites ]
+	);
+	const search: Partial< ViewTable | ViewGrid > = sitesRoute.useSearch();
```

Comment:
Why the type narrowing? What's wrong with `Partial< View >`?

Reply:
TS will complain that view.type can be `list` and that it's not available in default layouts, if I remember correctly.

Reply:
Right, makes sense. Same thing I ran into later.